src/server/main_service.cc

```diff
@@ -917,6 +921,11 @@ void Service::DispatchCommand(CmdArgList args, facade::ConnectionContext* cntx)
   }
 
   uint64_t start_ns = ProactorBase::GetMonotonicTimeNs(), end_ns;
+  double maxmemory_ratio = GetFlag(FLAGS_maxmemory_ratio);
+  uint64_t used_memory = ServerState::tlocal()->GetCachedUsedMemory(start_ns);
+  if (used_memory > (max_memory_limit * maxmemory_ratio) && (cid->opt_mask() & CO::DENYOOM)) {
```

Comment:
also please go over all write commands (there are not many of them) and check which one of them should have  DENYOOM - I think some have but they should they not and the opposite.

Reply:
I did this in another PR